lib/browser/api/net.js

```diff
@@ -244,7 +243,8 @@ function parseOptions (options) {
     }
   }
   if (options.session) {
-    if (options.session instanceof Session) {
+    // Weak check, but it should be enough to catch 99% of accidental misuses.
+    if (options.session.constructor.name === 'Session') {
```

Comment:
This check needs to be way safer.  `options.session.constructor` could be falsey / not an object